inst/include/child_model_simulation.hpp

```diff
@@ -674,88 +674,99 @@ void run_child_hiv_infections(int t,
-          (state_next.base.p_total_pop(2,s) - state_next.base.p_hiv_pop(2,s)) / ((state_next.base.p_total_pop(2,0) + state_next.base.p_total_pop(2,1)) - (state_next.base.p_hiv_pop(2,0) + state_next.base.p_hiv_pop(2,1)));
-        state_next.base.p_infections(2, s) += state_next.children.hiv_births  * cpars.hc1_cd4_dist(hd) * intermediate.children.bf_transmission_rate(3) *
-          (state_next.base.p_total_pop(2,s) - state_next.base.p_hiv_pop(2,s)) / ((state_next.base.p_total_pop(2,0) + state_next.base.p_total_pop(2,1)) - (state_next.base.p_hiv_pop(2,0) + state_next.base.p_hiv_pop(2,1)));
-      }// end hc1DS
-      //12-24
-      state_next.base.p_hiv_pop(1, s) += state_next.children.hiv_births * intermediate.children.bf_transmission_rate(2) *
-        (state_next.base.p_total_pop(1,s) - state_next.base.p_hiv_pop(1,s)) / ((state_next.base.p_total_pop(1,0) + state_next.base.p_total_pop(1,1)) - (state_next.base.p_hiv_pop(1,0) + state_next.base.p_hiv_pop(1,1)));
-      //24 plus
-      state_next.base.p_hiv_pop(2, s) +=  state_next.children.hiv_births  * (intermediate.children.bf_transmission_rate(3))*
-        (state_next.base.p_total_pop(2,s) - state_next.base.p_hiv_pop(2,s)) / ((state_next.base.p_total_pop(2,0) + state_next.base.p_total_pop(2,1)) - (state_next.base.p_hiv_pop(2,0) + state_next.base.p_hiv_pop(2,1)));
-    }// end NS
+    internal::run_bf_transmission_rate(t, pars, state_curr, state_next, intermediate, 12, ss_c.hBF, 3);
+    for (int s = 0; s < ss_b.NS; ++s) {
+      auto uninfected_prop_12_24 = (n_ba.p_total_pop(1, s) - n_ba.p_hiv_pop(1, s)) /
+                                   (n_ba.p_total_pop(1, 0) - n_ba.p_hiv_pop(1, 0) + n_ba.p_total_pop(1, 1) - n_ba.p_hiv_pop(1, 1));
+      auto uninfected_prop_24_plus = (n_ba.p_total_pop(2, s) - n_ba.p_hiv_pop(2, s)) /
+                                     (n_ba.p_total_pop(2, 0) - n_ba.p_hiv_pop(2, 0) + n_ba.p_total_pop(2, 1) - n_ba.p_hiv_pop(2, 1));
+      auto bf_hiv_transmission_12_24 = n_hc.hiv_births * i_hc.bf_transmission_rate(2) * uninfected_prop_12_24;
+      auto bf_hiv_transmission_24_plus = n_hc.hiv_births * i_hc.bf_transmission_rate(3) * uninfected_prop_24_plus;
+      for (int hd = 0; hd < ss_c.hc1DS; ++hd) {
+        // 12-24
+        n_hc.hc1_hiv_pop(hd, 3, 1, s) += p_hc.hc1_cd4_dist(hd) * bf_hiv_transmission_12_24;
+        n_ba.p_infections(1, s) += n_hc.hc1_hiv_pop(hd, 3, 1, s);
+
+        // 24 plus
+        auto new_hc1_infections_24_plus = p_hc.hc1_cd4_dist(hd) * bf_hiv_transmission_24_plus;
+        n_hc.hc1_hiv_pop(hd, 3, 2, s) += new_hc1_infections_24_plus;
+        n_ba.p_infections(2, s) += new_hc1_infections_24_plus;
+      } // end hc1DS
+
+      // 12-24
+      n_ba.p_hiv_pop(1, s) += bf_hiv_transmission_12_24;
+
+      // 24 plus
+      n_ba.p_hiv_pop(2, s) += bf_hiv_transmission_24_plus;
+    } // end NS
```

Comment:
this entire thing is worth checking in your editor, github diff makes it look really messy, i have introduced several variables and factorised/simplified a decent amount of algebra

Reply:
@mwalte10 -- do we have a test that would fail if something went awry in this refactor?

Reply:
something did go wrong in this refactor, the "Infections among children align" test in child model failed and i managed to fix it, but also some piece of mind, with all these tests, i run the child model on main, save that output object, then i save out the object from this branch with the child model running and i compare the two objects

the output is the exact same with a tolerance of 1e-10 (due to floating point errors)